packages/repository/src/relations/belongs-to/belongs-to.inclusion-resolver.ts

```diff
@@ -52,12 +52,13 @@ export function createBelongsToInclusionResolver<
     const sourceKey = relationMeta.keyFrom;
     const sourceIds = entities.map(e => (e as AnyObject)[sourceKey]);
     const targetKey = relationMeta.keyTo as StringKeyOf<Target>;
+    const dedupedSourceIds = deduplicate(sourceIds);
 
     const targetRepo = await getTargetRepo();
     const targetsFound = await findByForeignKeys(
       targetRepo,
       targetKey,
-      deduplicate(sourceIds),
+      dedupedSourceIds.filter(e => e),
```

Comment:
Is it possible to have a valid id value that's not truthy? For example, I can imagine `0` being a valid primary key. Your condition will filter such value out.